### PR TITLE
corrections for substation diagram

### PR DESCRIPTION
**Pull request by FranckLecuyer**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
The shunt cells might not be properly displayed in substation diagram, due to possible feeder cell orientation changed in smart layout, and also due to new bugs in the coordinate computation of the nodes inside this kind of cell type, and also due to recent refactoring in the block/cells position and coords computation.
Transformers svg symbols might not be properly displayed in substation diagram (pb with orientation and colorization), due to recent changes in the transformer representation in the component svg library.

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by floriand-e2r** [comment]
The two-winding transformer is correctly rotated in substation.svg, but alas! it's still suffering a 90° rotation in substation_no_feeder_values.svg and substation_optimized.svg

**Comment by floriand-e2r** [comment]
I got it, it's because the substation graph is "damaged" when generating the SVG file. In fact this explains the issues #178 and #177 that I just created. To "correct" it, the substation graph needs to be fresh new when writing the svg in the tests...
Was this already known?

**Comment by FranckLecuyer** [comment]
> I got it, it's because the substation graph is "damaged" when generating the SVG file. In fact this explains the issues #178 and #177 that I just created. To "correct" it, the substation graph needs to be fresh new when writing the svg in the tests...
> Was this already known?

I have just corrected that in this PR : Thanks for having seen that ....

**Review by floriand-e2r — CHANGES_REQUESTED** [review]
Could you also fill the PR description briefly?

File: single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramStyleProvider.java [hunk 1]
```diff
@@ -140,14 +140,44 @@ protected void addHighlightStateStyle(Edge edge, Map<String, String> style, Stri
...
-                Node n1 = adjacentNodes.get(0);
-                Node n2 = adjacentNodes.get(1);
+                adjacentNodes.sort(Comparator.comparingDouble(Node::getX));
+                FeederWithSideNode node1 = (FeederWithSideNode) adjacentNodes.get(0);
+                FeederWithSideNode node2 = (FeederWithSideNode) adjacentNodes.get(1);
+                FeederWithSideNode nodeWinding1 = node1.getSide() == FeederWithSideNode.Side.ONE ? node1 : node2;
+                FeederWithSideNode nodeWinding2 = node1.getSide() == FeederWithSideNode.Side.TWO ? node1 : node2;
+                FeederWithSideNode nodeWinding = nodeWinding1;
+
+                if (!node.isRotated() || Math.abs(node.getRotationAngle()) == 180.) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramStyleProvider.java]: Corrected

File: single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramStyleProvider.java [hunk 2]
```diff
@@ -140,14 +140,44 @@ protected void addHighlightStateStyle(Edge edge, Map<String, String> style, Stri
...
+                    if (subComponentName.equals(WINDING1)) {
+                        if (!node.isRotated()) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultDiagramStyleProvider.java]: Conditions changed

**Review by floriand-e2r — APPROVED** [review]

**Pull request merged**